client/js/uploader.basic.api.js

```diff
@@ -1067,6 +1089,35 @@
             });
         },
 
+        _handleDeleteSuccess: function(id) {
+            var name = this.getName(id);
+
+            this._netUploadedOrQueued--;
+            this._netUploaded--;
+            this._handler.expunge(id);
+            this._uploadData.setStatus(id, qq.status.DELETED);
+            this.log("Delete request for '" + name + "' has succeeded.");
+        },
+
+        _handleDeleteFailed: function(id, xhrOrXdr) {
+            var name = this.getName(id);
+
+            this._uploadData.setStatus(id, qq.status.DELETE_FAILED);
+            this.log("Delete request for '" + name + "' has failed.", "error");
+
+            // Check first if xhrOrXdr is actually passed since manual status changes don't handle this.
+            if (xhrOrXdr) {
```

Comment:
I think this section should be refactored to always call the `onError` callback, even if `xhrOrXdr` is absent, in which case the error message can simply omit any references to `xhrOrXdr`.

This may work (replaces everything starting w/ line 1108:

```js
// For error reporting, we only have access to the response status if this is not
// an `XDomainRequest`.
if (!xhrOrXdr || xhrOrXdr.withCredentials === undefined) {
    this._options.callbacks.onError(id, name, "Delete request failed", xhrOrXdr);
}
else {
    this._options.callbacks.onError(id, name, "Delete request failed with response code " + xhrOrXdr.status, xhrOrXdr);
}
```